src/core/server/bootstrap.ts

```diff
@@ -68,10 +65,13 @@ export async function bootstrap({
     isDevClusterMaster: isMaster && cliArgs.dev && features.isClusterModeSupported,
   });
 
-  const rawConfigService = new RawConfigService(
-    env.configs,
-    rawConfig => new LegacyObjectToConfigAdapter(applyConfigOverrides(rawConfig))
-  );
+  const rawConfigService = new RawConfigService(env.configs, rawConfig => {
+    if (features.stripXpackConfig) {
```

Comment:
don't have a strong opinion, but probably it's a good idea to keep all config manipulations in a single place https://github.com/restrry/kibana/blob/2a1493169d5e58a7112f3870067d1f4a7840d3e2/src/cli/serve/serve.js#L67